packages/common/src/downloads/getDownloadConfiguration.ts

```diff
@@ -0,0 +1,98 @@
+import { IHubAdditionalResource } from "../core/types/IHubAdditionalResource";
+import {
+  FlowType,
+  IDownloadFormatConfiguration,
+  IEntityDownloadConfiguration,
+  IHubEditableContent,
+} from "../core/types/IHubEditableContent";
+import { getProp } from "../objects/get-prop";
+import { IDynamicDownloadFormat } from "./types";
+import { getCreateReplicaFormats } from "./_internal/format-fetchers/getCreateReplicaFormats";
+import { getExportImageFormats } from "./_internal/format-fetchers/getExportImageFormats";
+import { getPagingJobFormats } from "./_internal/format-fetchers/getPagingJobFormats";
+import { getDownloadFlow } from "./_internal/getDownloadFlow";
+
+/**
+ * Returns the download configuration for an entity at this moment in time.
+ *
+ * If no configuration exists, a default configuration is returned based on the entity's current download flow.
+ * If a configuration exists but is no longer valid, the default configuration will also be returned.
+ *
+ * @param entity entity to get download configuration for
+ * @returns the current download configuration for the entity
+ */
+export function getDownloadConfiguration(
+  entity: IHubEditableContent
+): IEntityDownloadConfiguration {
+  // TODO: account for enterprise environments
+  const downloadFlow = getDownloadFlow(entity);
+  let serverFormats: IDynamicDownloadFormat[] = [];
+  const additionalResources: IHubAdditionalResource[] =
+    getProp(entity, "extendedProps.additionalResources") || [];
+  const existingConfiguration: IEntityDownloadConfiguration = getProp(
+    entity,
+    "extendedProps.downloads"
+  );
+
+  // TODO: Could we make a formal helper function instead?
+  // Maybe something like this:
+  //
+  // type LogicByFlowMap = Record<FlowType, () => void>;
+  // function executeLogicByFlow(
+  //  flow: FlowType,
+  //  logicByFlow: LogicByFlowMap
+  // )
+  // OR a switch statement: https://medium.com/technogise/type-safe-and-exhaustive-switch-statements-aka-pattern-matching-in-typescript-e3febd433a7a
+
+  const actionsByFlow: Record<FlowType, () => void> = {
+    createReplica: () => {
+      serverFormats = getCreateReplicaFormats(entity);
+    },
+    paging: () => {
+      serverFormats = getPagingJobFormats();
+    },
+    exportImage: () => {
+      serverFormats = getExportImageFormats(entity);
+    },
+  };
+
+  actionsByFlow[downloadFlow] && actionsByFlow[downloadFlow]();
```

Comment:
I would opt for the helper function like you said. It's a complex part of an already meaty function.